File: C++/Gas Station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int cur = 0, pos = 0;
        for (int i = 0; i < gas.size(); i ++) {
            cur += (gas[i] - cost[i]);
            if (cur < 0) pos = i + 1;
        }
        return pos;
    }
};

File: Python/Gas Station.py
class Solution:
    # @param gas, a list of integers
    # @param cost, a list of integers
    # @return an integer
    def canCompleteCircuit(self, gas, cost):
        if sum(gas) < sum(cost):
            return -1
        num = len(gas)
        diff = 0
        index = 0
        for i in range(num):
            if diff + gas[i] < cost[i]:
                diff = 0
                index = i + 1
            else:
                diff += gas[i] - cost[i]
        return index